src/lib/operators/abstract_operator.cpp

```diff
@@ -258,13 +273,38 @@ std::shared_ptr<const AbstractOperator> AbstractOperator::left_input() const { r
 std::shared_ptr<const AbstractOperator> AbstractOperator::right_input() const { return _right_input; }
 
 void AbstractOperator::set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {
-  DebugAssert(!_executed, "Setting parameters on operators that have already executed is illegal.");
+  Assert(_state == OperatorState::Created, "Setting parameters is allowed for OperatorState::Created only.");
   if (parameters.empty()) return;
   _on_set_parameters(parameters);
   if (left_input()) mutable_left_input()->set_parameters(parameters);
   if (right_input()) mutable_right_input()->set_parameters(parameters);
 }
 
+OperatorState AbstractOperator::state() const { return _state; }
+
+std::shared_ptr<OperatorTask> AbstractOperator::get_or_create_operator_task() {
+  std::lock_guard<std::mutex> lock(_operator_task_mutex);
+  // Return the OperatorTask that owns this operator if it already exists.
+  if (!_operator_task.expired()) return _operator_task.lock();
+
+  if constexpr (HYRISE_DEBUG) {
+    // Check whether _operator_task points to NULL, which means it was never initialized before.
+    using wt = std::weak_ptr<OperatorTask>;
```

Comment:
acr. ser. s.

Reply:
?

Reply:
Acronyms seriously suck.

Reply:
Thanks, now I get it. :D

I changed `wt` to `weak_null_pointer`.